DOC fix and clean up documentation links

zeaiso opened the pull request:
- Fixed relative paths in features section 
- Removed link to non-existent markdown file

zeaiso committed:
DOC fix and clean up documentation links

Fixed relative paths in features section
Removed link to non-existent markdown file

zeaiso committed:
DOC fix and clean up documentation links

Fixed relative paths in features section
Removed link to non-existent markdown file

GuySartorelli reviewed (changes requested):
Other than the one link to a file that no longer exists, these links are correct. The documentation is used to generate [this userhelp site](https://userhelp.silverstripe.org/en/5/optional_features/forms/) which has its own link handling functionality.

Strictly speaking the `.md` should be removed per [the contribution guidelines](https://docs.silverstripe.org/en/5/contributing/documentation/#links-to-documentation) so if you'd like to update them to match that style I'd be greatful.

Otherwise feel free to close this PR if you only wanted to update the links for viewing the markdown in GitHub as that isn't the recommended way to view them.

zeaiso committed:
DOC fix and adjust links based on contribution guidelines

- Adjusted links according to the contribution guidelines for documentation
- Removed link to non-existent markdown file

zeaiso commented:
@GuySartorelli  Thank you for the explanation, and I apologize for the PR! This was my first time using SilverStripe, and to understand how the user form works, I cloned it locally and reviewed the code in my IDE. Since I hadn’t checked the documentation online, I got a bit mixed up about the link structure. I thought updating the links might help others avoid similar issues, so I went ahead with the PR. Sorry for that mistake..

I’ve now updated the links based on the contribution guidelines.

GuySartorelli reviewed (approved):
Thanks for doing this! Sorry it took so long to review.
